### PR TITLE
Crow Rising

### PR DESCRIPTION
**Pull request by CrowRising**
Did not finish Iteration 3.
Needed a little more time to figure out how to get a hash as my return and not an array.
A little too much time in Iteration 2 not catching an error in my test.
Getting better, but not quite there.

### CODE REVIEW

File: lib/course.rb [hunk 1]
```diff
@@ -1 +1,22 @@
...
+  end
+
+  def full?
+    @full
+  end
+
+  def enroll(student)
+    @students << student
+    @full = true
+  end
```
Comment (jamisonordway) [inline lib/course.rb:21]: this will always set `@full` to true-- even if the capacity is 100 and we've only enrolled 1 student. I would remove line 20.

File: lib/course.rb [hunk 2]
```diff
@@ -1 +1,22 @@
...
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @students = []
+    @full = false
+  end
+
+  def full?
+    @full
+  end
```
Comment (jamisonordway) [inline lib/course.rb:16]: You already have attributes available that will let you check this. I don't think we need `@full` as an attribute at all:
```suggestion
  def full?
    @students.count >= @capacity
  end
```

File: spec/course_spec.rb [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      student1 = Student.new({name: "Morgan", age: 21})
+      student2 = Student.new({name: "Jordan", age: 29})
+
+      course.enroll(student1) 
+      course.enroll(student2) 
+
+      expect(course.students).to eq([student1, student2])
+      expect(course.full?).to be true
+
+    end
```
Comment (jamisonordway) [inline spec/course_spec.rb:29]: this test returns a false positive; be sure to also check for a false value returned from the `full?` method.
```suggestion
    it 'checks enrollment' do
      course = Course.new("Calculus", 2) 
      student1 = Student.new({name: "Morgan", age: 21})
      student2 = Student.new({name: "Jordan", age: 29})

      course.enroll(student1) 
      expect(course.full?).to be false
      course.enroll(student2) 

      expect(course.students).to eq([student1, student2])
      expect(course.full?).to be true

    end
```

File: lib/gradebook.rb [hunk 4]
```diff
@@ -1 +1,27 @@
 
+class GradeBook
+  attr_reader :instructor
+              :courses
+
+  def initialize(instructor)
+    @instructor = instructor
+    @courses = []
+    
+  end
```
Comment (jamisonordway) [inline lib/gradebook.rb:10]:
```suggestion
  def initialize(instructor)
    @instructor = instructor
    @courses = []
  end
```

File: spec/gradebook_spec.rb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      student1 = Student.new({name: "Morgan", age: 21})
+      student2 = Student.new({name: "Jordan", age: 29})
+
+      course.enroll(student1)
+      course.enroll(student2)
+      gradebook.list_all_students
+
+      expect(gradebook.list_all_students).to be_a Hash
+       
+
```
Comment (jamisonordway) [inline spec/gradebook_spec.rb:37]: Would like to see more tests here such as
```suggestion
      expect(gradebook.list_all_students).to be_a Hash
       
      expect(gradebook.list_all_students).to eq({course => [student1, student2]})
```

File: lib/gradebook.rb [hunk 6]
```diff
@@ -1 +1,27 @@
...
+  def courses
+    @courses
+  end
+
+  def add_course(course)
+    @courses << course
+  end
+
+  def list_all_students
+    students = {}
```
Comment (jamisonordway) [inline lib/gradebook.rb:21]: I think this method is pretty close to working and could be fixed by giving this new hash default values:
```suggestion
    students = Hash.new([])
```

File: spec/gradebook_spec.rb [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+      gradebook = GradeBook.new("Prof. Jerome")
+      course = Course.new("Calculus", 2)
+      student1 = Student.new({name: "Morgan", age: 21})
+      student2 = Student.new({name: "Jordan", age: 29})
+
+      course.enroll(student1)
+      course.enroll(student2)
+      gradebook.list_all_students
+
+      expect(gradebook.list_all_students).to be_a Hash
```
Comment (jamisonordway) [inline spec/gradebook_spec.rb:35]: Don't forget to add `course` to `gradebook`:
```
gradebook.add_course(course)
```

File: spec/course_spec.rb (hunk 3 above)
Comment (CrowRising) [reply spec/course_spec.rb:29]: That makes sense. I will pay attention to testing edge cases (if that is what this was called)

File: lib/course.rb (hunk 2 above)
Comment (CrowRising) [reply lib/course.rb:16]: Ahhhh, yes. The poor decisions made in the panic of the timed test.

File: lib/course.rb (hunk 1 above)
Comment (CrowRising) [reply lib/course.rb:21]: duh!!! Okay yes that makes sense. Thank you
